activerecord/test/cases/attribute_methods_test.rb

```diff
@@ -435,6 +435,17 @@ def topic.title() "b" end
     assert_equal true, Topic.new(author_name: "Name").author_name?
   end
 
+  test "user defined text attribute predicate" do
+    klass = Class.new(ActiveRecord::Base) do
+      self.table_name = Topic.table_name
+
+      attribute :user_defined_text, :text
+    end
+
+    topic = klass.new(user_defined_text: "text")
+    assert_predicate topic, :user_defined_text?
```

Comment:
Probably this assertion will pass without this PR.
Can you add more assertions for `"f"`, `"false"`, `"off"`, or `ActiveModel::Type::Boolean::FALSE_VALUES`, etc.